addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3876,6 +3883,7 @@ var SnippetsMenu = Widget.extend({
         for (const gridItemEl of gridItemEls) {
             gridUtils._reloadLazyImages(gridItemEl);
         }
+        await this._updateInvisibleDOM();
```

Comment:
What `_updateInvisibleDOM` does seems already mutex-protected. As awaiting this would not impact anything as inside an event handler, I would not await. At best I would add a comment like `// This is async but using the main editor mutex` or something like that.